src/rapidfuzz/process_cpp.py

```diff
@@ -85,3 +86,35 @@ def cdist(
             **kwargs,
         )
     )
+
+
+def cpdist(
+    queries: Collection[Sequence[Hashable] | None],
+    choices: Collection[Sequence[Hashable] | None],
+    *,
+    scorer: Callable[..., int | float] = ratio,
+    processor: Callable[..., Sequence[Hashable]] | None = None,
+    score_cutoff: int | float | None = None,
+    score_hint: int | float | None = None,
+    score_multiplier: int | float = 1,
+    dtype: np.dtype | None = None,
+    workers: int = 1,
+    **kwargs: Any,
+) -> np.ndarray:
+    import numpy as np
+
+    dtype = _dtype_to_type_num(dtype)
+    return np.asarray(
```

Comment:
I think in the long run I would want both `cdist` and `cpdist` to return the custom matrix / vector, so people can use these parts without numpy if they have less complex use cases. While everyone else can continue to use `np.asarray` to get the full feature set of numpy without any copies, since it works on the same internal buffer.

This would be a breaking change for version 4 though, so this is fine for now. I think this should return an array instead of a matrix though.

Reply:
I presume you mean the output should have shape (n,) not (n,1). I have flattened the output to do this.